src/model/CoilCoolingWater.cpp

```diff
@@ -273,48 +273,68 @@ namespace model {
+          return false;
+        }
+      }
 
-        success = WaterToAirComponent_Impl::addToNode(node);
-        auto t_containingZoneHVACComponent = containingZoneHVACComponent();
+      success = WaterToAirComponent_Impl::addToNode(node);
+      if (!success) {
+        return false;
+      }
+      // We only add the controller if we are not part of a containing ZoneHVACComponent, and we have a PlantLoop assigned
+      if (containingZoneHVACComponent() || !waterInletModelObject()) {
+        // We're done
+        return true;
+      }
 
-        if (success && (!t_containingZoneHVACComponent)) {
-          if (auto t_waterInletModelObject = waterInletModelObject()) {
-            if (auto oldController = controllerWaterCoil()) {
-              if (!openstudio::istringEqual(oldController->action().get(), "Reverse")) {
-                LOG(Warn,
-                    briefDescription()
-                      << " has an existing ControllerWaterCoil with action set to something else than 'Reverse'. Make sure this is what you want");
-              }
-            } else {
-              if (t_containingHVACComponent && t_containingHVACComponent->optionalCast<CoilSystemCoolingWater>()) {
-                // no-op
-              } else {
-                Model t_model = model();
-                ControllerWaterCoil controller(t_model);
-                controller.getImpl<ControllerWaterCoil_Impl>()->setWaterCoil(getObject<HVACComponent>());
-                controller.setAction("Reverse");
-              }
+      // We don't need a controller if we are part of a CoilSystemCoolingWater (directly, or indirectly via a
+      // CoilSystemCoolingWaterHeatExchangerAssisted)
+      bool needAController = true;
+      if (t_containingHVACComponent) {
+        if (t_containingHVACComponent->optionalCast<CoilSystemCoolingWater>()) {
+          needAController = false;
+        } else if (auto hxAssisted_ = t_containingHVACComponent->optionalCast<CoilSystemCoolingWaterHeatExchangerAssisted>()) {
+          if (auto parent_ = hxAssisted_->containingHVACComponent()) {
+            if (parent_->optionalCast<CoilSystemCoolingWater>()) {
+              needAController = false;
             }
           }
         }
       }
 
+      if (auto oldController_ = controllerWaterCoil()) {
+        if (!needAController) {
+          oldController_->remove();
+        } else if (!openstudio::istringEqual(oldController_->action().get(), "Reverse")) {
+          LOG(Warn, briefDescription()
+                      << " has an existing ControllerWaterCoil with action set to something else than 'Reverse'. Make sure this is what you want");
+        }
+      } else if (needAController) {
+        Model t_model = model();
+        ControllerWaterCoil controller(t_model);
+        controller.getImpl<ControllerWaterCoil_Impl>()->setWaterCoil(getObject<HVACComponent>());
+        controller.setAction("Reverse");
+      }
```

Comment:
If we need a controller:
* If there is one, we kept it, but warn if not reverse
* If there isn't one: we create it

If we don't need a controller, just remove any existing one